src/metadataExt.ts

```diff
@@ -0,0 +1,261 @@
+     * @param {ICopyDashboardOptions} options object with options:
+     *      - keepReferences {boolean} choose whether dashboard is cloned with same references
+     *      - name {string} new name for dashboard
+     * @return {string} uri of cloned dashboard
+     */
+    public async saveAs(
+        projectID: string,
+        dashboardUri: string,
+        options: ICopyDashboardOptions,
+    ): Promise<string> {
+        const dashboard = await this.metadataModule.getObjectDetails(dashboardUri);
+        const { analyticalDashboard } = await this.metadataModule.createObject(projectID, {
+            ...dashboard,
+            analyticalDashboard: {
+                ...dashboard.analyticalDashboard,
+                meta: {
+                    ...dashboard.analyticalDashboard.meta,
+                    title: this.getDashboardName(dashboard.analyticalDashboard.meta.title, options.name),
+                },
+            },
+        });
+
+        const dashboardID: string = analyticalDashboard.meta.uri.split("/")[5];
+        const types = ["visualizationObject", "kpi", "visualizationWidget", "filterContext"];
+        const objectsUsingDashboard: any[] = await this.getObjectsUsingDashboard(projectID, dashboardUri, {
+            types,
+        });
+
+        if (options.keepReferences) {
+            return analyticalDashboard.meta.uri;
+        } else {
+            const clonedDashboard = await this.updateWidgets(
+                projectID,
+                analyticalDashboard,
+                dashboardID,
+                objectsUsingDashboard,
+            );
+
+            return clonedDashboard;
+        }
+    }
+
+    /**
+     * Deletes dashboard and its objects
+     * (only the author of the dashboard can delete the dashboard and its objects)
+     *
+     * @method deleteAllObjects
+     * @param {string} projectId Project identifier
+     * @param {string} dashboardUri Uri of a dashboard to be deleted
+     */
+
+    public async cascadingDelete(projectID: string, dashboardUri: string): Promise<any> {
+        const objects = await this.metadataModule.getObjectUsing(projectID, dashboardUri);
+        const currentUserInfo = await (await this.userModule.getAccountInfo()).profileUri;
+
+        const objectsToBeDeleted = objects
+            .filter((object: any) => {
+                return object.author === currentUserInfo;
+            })
+            .map((object: any) => {
+                return object.link;
+            });
+
+        this.xhr.post(`/gdc/md/${projectID}/objects/delete`, {
+            body: {
+                delete: {
+                    items: [dashboardUri].concat(objectsToBeDeleted),
+                    mode: "cascade",
+                },
+            },
+        });
+    }
+
+    private getDashboardName(originalName: string, newName?: string) {
+        if (newName !== undefined) {
+            return newName;
+        }
+        return `copyOf(${originalName})`;
```

Comment:
the default name should be less 'techy'.  i propose minor reword 'Copy of ${originalName}'